websites/W/Wolfy/presence.ts

```diff
@@ -93,45 +102,58 @@ async function handleCheckingLeaderboard(
 
 	if (!username) {
 		const classementType =
-			document.querySelector("div.Leaderboard_moonLeaderboard__3U2H7") !== null
+			document.querySelector("div.Leaderboard_moonLeaderboard__7q9CE") !== null
 				? "Lunaire"
-				: document.querySelector("div.Leaderboard_lifeLeaderboard__3Wtz1") !==
+				: document.querySelector("div.Leaderboard_lifeLeaderboard__itvcT") !==
 				  null
 				? "Éternel"
 				: null;
 
 		presenceData.details = `Regarde le classement ${classementType}`;
 		presenceData.state = `Top ${parseInt(
 			document.querySelector(
-				"div.Leaderboard_playerLine__1uAgP > div.Leaderboard_rank__n6aio"
+				"div.Leaderboard_playerLine__Qo9eG > div.Leaderboard_rank__n6Cz_"
 			)?.textContent
 		).toLocaleString()} ${classementType} (${parseInt(
-			document.querySelector("div.Leaderboard_lifetimeXp__372DW > p")
+			document.querySelector("div.Leaderboard_lifetimeXp__UpQAh > p")
 				?.textContent
 		).toLocaleString()} ${classementType === "Lunaire" ? "lauriers" : "xp"})`;
 	} else {
 		presenceData.details = `Regarde le profil de ${username}`;
 		presenceData.state = `[${
-			document.querySelector("div.ExperienceGroup_first__3h_RY > p")
+			document.querySelector("div.ExperienceGroup_first__cLc5y > p")
 				?.textContent
 		}] ${parseInt(
 			document
-				.querySelector("div.ExperienceGroup_experienceBarFull__3Qo8A > span")
+				.querySelector("div.ExperienceGroup_experienceBarFull__xUSsB > span")
 				?.textContent.split(" / ")[0]
 		).toLocaleString()} xp & ${parseInt(
-			document.querySelector("p.PlayerCard_number__1d0CM").textContent
+			document.querySelector("p.PlayerCard_number__BC5Cv").textContent
 		).toLocaleString()} lauriers`;
 		await addVisitProfilButton(presenceData, username);
 	}
 }
 
+function isWaitingGame(state: string, lang: string) {
+	return state === waitingString[resolveLanguage(lang)];
+}
+
+function resolveLanguage(lang: string): keyof typeof waitingString {
+	if (Object.keys(waitingString).includes(lang))
+		return lang as keyof typeof waitingString;
+	return "en";
+}
+
 presence.on("UpdateData", async () => {
 	const presenceData: PresenceData = {
 		largeImageKey: "logo"
 	};
 
 	path = document.location.pathname;
 
+	const pathOffset =
+		path.split("/")[1] === document.querySelector("html")?.lang ? 1 : 0; // If the language is the first path element, we need to add an offset
```

Comment:
instead of doing this, you should make a constant in the event for `path.split("/")` to avoid computing it time and time again and pop out the element you need to pop out, I believe that's more efficient

Reply:
Idk if this was what you was talking about but i just did it in commit [c110352](https://github.com/PreMiD/Presences/pull/6224/commits/c1103523cb526056aab767695a51997f623351f6)